developer-docs-site/static/messageboard_demo/client/README.md

```diff
@@ -0,0 +1,187 @@
+}
+```
+
+When a participant want to change the pinned message, the participant first need to borrow the board resource under creator’s account and check if the participant is in the ACL. Then it borrows mutable board to modify the pinned_post. 
+
+```tsx
+public(script) fun send_pinned_message(
+    account: &signer, board_addr: address, message: vector<u8>
+) acquires ACLBasedMB, MessageChangeEventHandle {
+    let board = borrow_global<ACLBasedMB>(board_addr);
+    assert!(ACL::contains(&board.participants, Signer::address_of(account)),EACCOUNT_NOT_IN_ACL);
+
+    let board = borrow_global_mut<ACLBasedMB>(board_addr);
+    board.pinned_post = message;
+```
+
+## Interacting With Messageboard
+
+### **Compile and Publish the Modules**
+
+The [following code](https://github.com/aptos-labs/aptos-core/blob/main/developer-docs-site/static/messageboard_demo/client/index.ts) basically runs a command to compile the messageboard contract with the account address of developer. 
+
+It then read the compiled modules in mv files in the module path and creates a txn to publish the modules under the developer’s account address.
+
+```tsx
+chdir('../../../../aptos-move/move-examples/messageboard');
+execSync(
+    `aptos move compile --package-dir . --named-addresses MessageBoard=${messageboardDev.address().toString()}`
+);
+var module_path = "build/MessageBoard";
+MessageboardUtil.installMessageboard(client, messageboardDev, module_path)
+```
+
+### **Interact with the Messageboard**
+
+We use [Aptos SDK](https://github.com/aptos-labs/aptos-core/tree/main/ecosystem/typescript/sdk) to interact with Aptos Chain. 
+
+After the modules are published under the developer’s account address, we can use the functions of these published modules to specify the function name, argument to create the messageboard initialization payload and send the txn to devnet to create the messageboard 
+
+```tsx
+var fname = `${this.contractAddr.toString()}::${this.boardType}::message_board_init`;
+var args = [];
+const initPayload = MessageboardUtil.getScriptFunctionTxnPayload(fname, args);
+await MessageboardUtil.executeTransaction(this.client, account, initPayload);
+```
+
+We can sendMessage to a messageboard through composing txn with the send_message_to payload
+
+```tsx
+async sendMessage(account: AptosAccount, message: string) {
+    var hexstring = Buffer.from(message).toString('hex');
+    var args = [this.adminAddr.toString(), hexstring];
+    var fname = `${this.contractAddr.toString()}::${this.boardType}::send_message_to`;
+    await MessageboardUtil.executeTransaction(this.client,
+        account,
+        MessageboardUtil.getScriptFunctionTxnPayload(fname, args)
+    );
+}
+```
+
+We can also read the events to query the events published to an account. by default, we return all the events. We can record the sequence number of last event we received. So we can read all the new events sent after our last recorded event.  
```

Comment:
We can also read the events by querying the events published to an account. By default, we return all the events. We can record the sequence number of last event we received. Hence we can read all the new events sent after our last recorded event.